helper/schema/schema.go

```diff
@@ -198,6 +198,14 @@ type Schema struct {
 	AtLeastOneOf  []string
 	RequiredWith  []string
 
+	// ConditionsMode specifies the behavior of the SDK when one of the
+	// above conditions are met. (ConflictsWith, ExactlyOneOf, etc).
+	// When set to WARNING, a warning message is displayed to the user.
+	// When set to ERROR (default), a fatal error is returned.
+	// Optionally, a message can be returned with the warning or error.
+	ConditionsMode    string
```

Comment:
Is there any chance we could make this an enum with constants, rather than a string?

E.g.:

```go
type SchemaConditionsMode uint8

const (
  SchemaConditionsModeError SchemaConditionsMode = 0
  SchemaConditionsModeWarning SchemaConditionsMode = 1
)
```

Then in this Schema struct it would be:

```go
type Schema struct {
  // everything else is the same, omitting
  ConditionsMode SchemaConditionsMode
  // everything else is the same, omitting
}
```

and in when using it, you'd do:

```go
mySchema := &schema.Schema{
  // everything else is the same, omitting
  ConditionsMode: schema.SchemaConditionsModeWarning,
  // everything else is the same, omitting
}
```

This has the neat benefit that we can define the default behavior (unset/0) as `SchemaConditionsModeError`, meaning we don't need to detect when it's unset, the default is just an error.

Reply:
Alternatively, we could make `ConditionsMode` accept a `diag.Severity`, but I think I like that less.

Reply:
Wow, that is really neat! Sure, I'll try this.